Dockerfile.testing

```diff
@@ -1,24 +1,21 @@
 # For building gcr.io/forgerock-io/secret-agent-testing:latest
```

Comment:
Also thoughts on using one dockerfile with a testing stage?

Reply:
The kubebuilder installation is about 285MB. It's need for testing, but not for production. Is there a way to produce two separate images using multi-stage builds?

Reply:
yea - you can use `docker build --target testing`

Reply:
Sweet! I didn't know that one. Using one Dockerfile works for me. 👍

Reply:
I don't think they're exactly compatible. The release Dockerfile is built with a separate stage that builds the go executable then copies that to the final stage.  The testing image does not really need the built package. 

Also, this wont work if we want the standard docker build to be the "release" build. `--target` stops the build when it reaches a specific build stage.  The testing image needs go binaries,  kubebuilder, git, etc (one extra stage) . So the "release" build is the one that will require `--target=release`

`docker build --target release -t myimage`  <<< builds the release image
`docker build -t myimage`  <<< builds the test image

Reply:
I tend to agree it would be a bit awkward have to build the non-testing image with the --target flag. Maybe @maxres-fr has an idea, but otherwise I see the point in having two Dockerfiles.

Reply:
@maxres-fr WDYT?

Reply:
Ok we talked about this on the :phone: we are going to leave as is for now, since the testing stage would be built and discarded during the build process for a release. Moving the release stage to the bottom of the file would make it the default target. The only advantage to the stage being left in would be running tests once again before completing the build.

Reply:
I created #36 to track this. I have a sample Dockerfile. It may/may not be a good idea. But let's see.